feat(downloads): allow streaming downloads access to response iterator (#1956)
python-gitlab/python-gitlab: docs/gl_objects/pipelines_and_jobs.rst

    subprocess.run(["unzip", "-bo", zipfn])
    os.unlink(zipfn)

> It is also possible to use the underlying iterator directly::

Reviewer: ```suggestion
Or, you can also use the underlying response iterator directly::
```

Author: Agreed,  that sounds better

    artifact_bytes_iterator = build_or_job.artifacts(streamed=True, action='iterator')

> This can be used with FastAPI/Starlette StreamingResponse to forward a download from gitlab without having to download
> the entire file server side first::

Reviewer: Maybe this would sound slightly more generic?

```suggestion
This can be used with frameworks that expect an iterator (such as FastAPI/Starlette's
``StreamingResponse``) to forward a download from GitLab without having to download
the entire content server-side first::
```

Author: Same here!
